src/ResizeObserverSPI.js

```diff
@@ -117,16 +157,33 @@ export default class ResizeObserverSPI {
         }
 
         const observations = this.observations_;
+        const observation = observations.get(target);
 
         // Do nothing if element is not being observed.
-        if (!observations.has(target)) {
+        if (!observation) {
             return;
         }
 
         observations.delete(target);
 
-        if (!observations.size) {
-            this.controller_.removeObserver(this);
+        if (this.intersectionObserver_) {
+            this.intersectionObserver_.unobserve(target);
+        }
+
+        // Disconnect the root if no longer used.
+        const {rootNode} = observation;
+        const rootNodeTargets = this.rootNodes_.get(rootNode);
+
+        if (rootNodeTargets) {
+            const index = rootNodeTargets.indexOf(target);
+
+            if (~index) {
```

Comment:
TIL about using tilde operator! Am I right in thinking this is equivalent to `if(index == -1) { `?

Reply:
`~-1 === 0`. This is the pattern this library is using, so I decided to keep it as is.